### PR TITLE
HIVE-25040: Drop database cascade cannot remove persistent functions

### PR DESCRIPTION
**Pull request by mustafaiman**
Change-Id: I9d32534f752650c23333432d304ad69a2f2cd693

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/exec/Registry.java [hunk 1]
```diff
@@ -553,21 +553,19 @@ private void addFunction(String functionName, FunctionInfo function) {
         Integer refCount = persistent.get(functionClass);
         persistent.put(functionClass, Integer.valueOf(refCount == null ? 1 : refCount + 1));
       }
+    } catch (ClassNotFoundException e) {
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/Registry.java:556]: The issues' root should be removePersistentFunctionUnderLock when trying to get per UDFClass below --  not sure why this is needed here as part of addFunction?

Comment (mustafaiman) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/Registry.java:556]: getPermanentUdfClass may throw ClassNotFoundException

Comment (pgaref) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/Registry.java:556]: I see, but this is not part of the Drop cascade issue right?

Comment (pgaref) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/Registry.java:556]: got it, just realized its been called from here

**Review by pgaref — APPROVED** [review]
Thanks Mustafa, +1 from my side, pending tests

**Pull request closed without merging**